src/libraries/System.Private.CoreLib/src/System/Reflection/NullabilityInfoContext.cs

```diff
@@ -452,7 +427,7 @@ private void TryLoadGenericMetaTypeNullability(MemberInfo memberInfo, Nullabilit
 
             if (metaType != null)
             {
-                CheckGenericParameters(nullability, metaMember!, metaType);
+                CheckGenericParameters(nullability, metaMember!, metaType, memberInfo.ReflectedType);
```

Comment:
ReflectedType is important for fixing https://github.com/dotnet/runtime/issues/63555